pxr/usd/plugin/usdAbc/alembicReader.cpp

```diff
@@ -894,22 +878,58 @@ _ReaderContext::_ReaderContext() :
 }
 
 bool
-_ReaderContext::Open(const std::string& filePath, std::string* errorLog)
+_ReaderContext::Open(const std::string& filePath, std::string* errorLog,
+                     const SdfFileFormat::FileFormatArguments* args)
 {
     Close();
 
-    IArchive archive;
+    std::vector<std::string> layeredABC;
+    if (args) {
+        const auto abcLayers = args->find("abcLayers");
+        if (abcLayers != args->end()) {
+            for (auto&& l : TfStringSplit(abcLayers->second, ",")) {
+                layeredABC.emplace_back(std::move(l));
+            }
+        }
+    }
+    layeredABC.emplace_back(filePath);
+
+#if PXR_HDF5_SUPPORT_ENABLED && !H5_HAVE_THREADSAFE
+    // HDF5 may not be thread-safe.
+    using lock_guard = std::lock_guard<std::recursive_mutex>;
+    std::unique_ptr<std::lock_guard<std::recursive_mutex>> hfd5Lock(new lock_guard(*_hdf5));
```

Comment:
Does the lock really need to be held for the call to getArchive() below?  If so, that's disappointing that Alembic doesn't take care of it itself, and _might_ lead to some performance regression in multithreaded stage-opening since we're now holding the lock while checking what the archive type is, but it might be pretty minor and the simplification is probably worth it.

If **not**, then could we start out with the unique_ptr as empty and reset it to a new lock_guard below only if we need to?

Reply:
This isn't exactly a code-path that's going to be exercised a lot here, but the problem is IFactory::getArchive may use HDF5 API's of which I'm not entirely clear are thread-safe or not.

As HDF5 support is optional in both USD and Alembic, I'd vote it's up to the client to make sure it's configured in the most optimal way (i.e. when **H5_HAVE_THREADSAFE=1** or **PXR_HDF5_SUPPORT_ENABLED=0**, there's no lock at all).

@lamiller0 suggested if you want to remove the lock entirely he'd be willing to handle it on the Alembic side...in the future...with a concrete case...whenever anyone encounters a problem.

Reply:
Ok, I’m fine leaving the pattern as-is, and taking @lamiller0 up on his kind offer only if it proves problematic. We don’t even have the datasets to be able to profile in vtune, so we’d rely on you or other heavy Alembic users to keep an eye out for stage-opening performance!